nextcord/embeds.py

```diff
@@ -427,17 +455,79 @@ def set_image(self: E, url: MaybeEmpty[Any]) -> E:
         -----------
         url: :class:`str`
             The source URL for the image. Only HTTP(S) is supported.
+        file: Optional[File]
+            A file to use for the image.
+
+            .. versionadded:: 2.0
         """
+        from . import InvalidArgument
 
-        if url is EmptyEmbed:
+        if url is EmptyEmbed and not file:
```

Comment:
This might need to be a separate if-block, since you can set the image to a url after setting it to a file for example, in which case it should delete only the file one.